.editorconfig

```diff
@@ -0,0 +1,6 @@
+root = true
+
+[*]
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = true
```

Comment:
I would strongly suggest making specific cases for VS build files.
The IDE can and will overwrite them with its own baked-in conventions (CRLF,  no ending newline).

You could add something like:
```
[*.{vcxproj,filters,vcxitems}]
insert_final_newline = false
end_of_line = crlf
tab_width = 2

[*.sln]
insert_final_newline = false
end_of_line = crlf
indent_style = tab
tab_width = 4

Reply:
Would that be better handled in the EditorConfig or the .gitattributes? I don't use Visual Studio that much to know what changes it makes by default.

or would it be better if we did it in both perhaps?

Reply:
Agreed. We should probably apply these rules in both files.
It's a nag to duplicate the work, but this repo will very unlikely need frequent updates of those rules.

To answer you question, I do think EditorConfig should drive these rules because it runs before commit time, keeping the sources consistent at save time.

Reply:
Ah, OK. Will do! Thanks for the feedback! 😄